### DISCUSSION TITLE
light weight version for raspberry?

### DISCUSSION THREAD

**Discussion by psy0rz** [zauberzeug/nicegui#2796] [Ideas / Feature Requests]
### Question

I love nicegui, but on a raspberry pi Zero W it takes like 20 seconds to load and start.

Is there a way to disable some functionality to make it faster?

Also  installing libsass is near impossible since it needs so much memory, but this is a known python3-libsass issue.

And i assume nicegui cant run without libsass?

Thanks

**Comment by falkoschindler** [comment]
Thanks for bringing this up, @psy0rz!
I wasn't aware of libsass being such a hassle on some platforms. We could probably make it an optional dependency.
There's also an [environment variable](https://nicegui.io/documentation/section_configuration_deployment#environment_variables) `MATPLOTLIB` that can be used to exclude the potentially costly import of matplotlib. Have you tried it?
Beyond that, I'm not sure what exactly takes so much time to load and what would be worth to make optional. We're open for ideas.

**Reply by rodja** [reply]
> We could probably make it an optional dependency.

I just created pull request #2809 for this.

**Comment by psy0rz** [comment]
thanks for the quick reply! 

i've seen you also have an excellent docker image, so i'm trying to use that now with podman on a raspberry.  

i'll try the MATHLIB suggestion as well.

**Comment by psy0rz** [comment]
MATPLOTLIB doesnt seem to help (same startup time /memory)

However, after measuring it, its more like 10 seconds, so maybe i was off in my first post. (or newer nicegui is faster?)

Also podman on the raspberry has barely any overhead, so i will use that from now on.  (And also distribute my project via a docker iamge when its ready)

Thanks!

**Comment by falkoschindler** [comment]
@psy0rz We're trying to make libsass optional (see PR #2809). But I wonder how bad this dependency really is. On what distribution are you experiencing the memory issues? And what Python version? Is there a bug report somewhere online? I only found pretty old articles and discussions and want to make sure we're not spending too much effort on obsolete problems. Thanks a lot!

**Comment by psy0rz** [comment]
well, i've already worked around it as i've mentioned above, via docker containers, which is my preferred method anyway.

some info:
```
root@psypi3:~# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy


root@psypi3:~# cat /proc/meminfo 
MemTotal:         424996 kB

```
